queries/rescript/highlights.scm

```diff
@@ -0,0 +1,340 @@
+(comment) @comment @spell
+
+; Identifiers
+;------------
+; Escaped identifiers like \"+."
+((value_identifier) @constant.macro
+  (#match? @constant.macro "^\\.*$"))
+
+(value_identifier) @variable
+
+[
+  (type_identifier)
+  (unit_type)
+  (list)
+  (list_pattern)
+] @type
+
+((type_identifier) @type.builtin
+  (#any-of? @type.builtin "int" "char" "string" "float" "bool" "unit"))
+
+[
+  (variant_identifier)
+  (polyvar_identifier)
+] @constructor
+
+(record_type_field
+  (property_identifier) @property)
+
+(record_field
+  (property_identifier) @property)
+
+(object
+  (field
+    (property_identifier) @property))
+
+(object_type
+  (field
+    (property_identifier) @property))
+
+(module_identifier) @module
+
+(member_expression
+  (property_identifier) @variable.member)
+
+(value_identifier_path
+  (module_identifier)
+  (value_identifier) @variable)
+
+(record_pattern
+  (value_identifier_path
+    (value_identifier) @variable.member))
+
+(record_pattern
+  (value_identifier) @variable)
+
+(labeled_argument
+  label: (value_identifier) @variable.parameter)
+
+; Parameters
+;----------------
+(list_pattern
+  (value_identifier) @variable.parameter)
+
+(spread_pattern
+  (value_identifier) @variable.parameter)
+
+; String literals
+;----------------
+[
+  (string)
+  (template_string)
+] @string
+
+(character) @character
+
+(escape_sequence) @string.escape
+
+; Other literals
+;---------------
+[
+  (true)
+  (false)
+] @boolean
+
+(number) @number
+
+(polyvar) @constructor
+
+(polyvar_string) @constructor
+
+; Functions
+;----------
+; parameter(s) in parens
+(parameter
+  (value_identifier) @variable.parameter)
+
+(labeled_parameter
+  (value_identifier) @variable.parameter)
+
+; single parameter with no parens
+(function
+  parameter: (value_identifier) @variable.parameter)
+
+; first-level descructuring (required for nvim-tree-sitter as it only matches direct
+; children and the above patterns do not match destructuring patterns in NeoVim)
```

Comment:
👀 
```suggestion
; first-level destructuring (required for Neovim as it only matches direct
; children and the above patterns do not match destructuring patterns)
```

👀  Can you comment on that?

Reply:
Good catch, think I will just remove that comment. Looking through the queries+grammar looks like this comment is basically just trying to say that those queries are necessary because, by default, something like `(parent (child) @cap)` uses immediate parent matching and not something like `has-ancestor` to match `child`. But that goes without saying and this comment just serves to increase confusion I think